wire/msgtx.go

```diff
@@ -1701,19 +1662,16 @@ func writeTxInWitnessValueSigning(w io.Writer, pver uint32, version int32,
 // readTxOut reads the next sequence of bytes from r as a transaction output
 // (TxOut).
 func readTxOut(r io.Reader, pver uint32, version int32, to *TxOut) error {
-	var buf8 [8]byte
-	_, err := io.ReadFull(r, buf8[:])
+	value, err := binarySerializer.Uint64(r, littleEndian)
```

Comment:
I had thought that readElement caused an extra alloc, or at least spent an extra check for which type was which before reading into it. So, all readElements were changed to use the serializer instead.